pkg/controller/propagator/propagation.go

```diff
@@ -432,27 +444,36 @@ func processTemplates(replicatedPlc *policiesv1.Policy, decision appsv1.Placemen
 			resolveddata, tplErr := tmplResolver.ResolveTemplate(policyT.ObjectDefinition.Raw, templateContext)
 			if tplErr != nil {
 				reqLogger.Error(tplErr, "Failed to resolve templates")
-				return tplErr
+
+				r.recorder.Event(rootPlc, "Warning", "PolicyPropagation",
```

Comment:
Can you make it an error rather than a warning?

Reply:
same as above ,  rest of the comments are resolved